pkgs/test_core/lib/src/runner/vm/platform.dart

```diff
@@ -54,6 +54,9 @@ class VMPlatform extends PlatformPlugin {
       isolate =
           await _spawnIsolate(path, receivePort.sendPort, suiteConfig.metadata);
       if (isolate == null) return null;
+      var onExitPort = ReceivePort();
+      _isolateExits.add(Result.capture(onExitPort.first));
+      isolate.addOnExitListener(onExitPort.sendPort);
```

Comment:
Are we concerned about races here if the isolate spawns + completes before we can register the listener? Ideally we'll want to pass this port directly to `Isolate.spawnUri`'s `onExit` to ensure we're registered before the isolate runs.

Reply:
No I don't think we need to be concerned. We control the initial behavior of the spawned Isolate and it won't exit until after it has run tests.

I thought about plumbing it through to the `Isolate.spawn` calls to be safer, but it adds a good deal of noise so I skipped it.

@jakemac53 - let me know if you think I'm overlooking a case where the isolate could immediately exit.

Reply:
If you have an error in your `main` function it could presumably immediately exit?

Reply:
I don't think it could. The `main` that runs is not the user `main`, it is ours and should not have any immediate errors. If an error happens in the user's main it will still communicate that back through the `SendPort`.

https://github.com/dart-lang/test/blob/9c0ae32d1e6771a43f37e0b2a7f5bd75abe51fd3/pkgs/test_api/lib/src/backend/remote_listener.dart#L135

I can go forward with the earlier `onExit` bind though for safety.

Reply:
If we did have an immediate error I also think the effects likely wouldn't be a problem. It would mean a second error is surfaced, but shouldn't lead to a hang. The `RecievePort` will end and the `.first` call should throw an error.

Reply:
> I don't think it could. The `main` that runs is not the user `main`, it is ours and should not have any immediate errors. If an error happens in the user's main it will still communicate that back through the `SendPort`.

Oh right good point.

We have had errors in the bootstrapping code before though and this would still help for that case (its notoriously hard to debug).

Reply:
@bkonyi - passing to the `onExit` listener seems to keep the isolate open in the cases where the isolate failed to spawn. Is there some way to prevent that? I tried also closing the `ReceivePort` in that case, but ti doesn't seem to help.

Reply:
Ah my mistake, I found another error handling path where the `RecievePort` was not getting close.